miq_scripts/run.rb

```diff
@@ -1,4 +1,13 @@
-print "Assinging alert profiles to the Enterprise\n"
+puts "Assinging alert profiles to the Enterprise"
 
 MiqAlertSet.find_by(:guid=>"a16fcf51-e2ae-492d-af37-19de881476ad").assign_to_objects(MiqEnterprise.last)
 MiqAlertSet.find_by(:guid=>"ff0fb114-be03-4685-bebb-b6ae8f13d7ad").assign_to_objects(MiqEnterprise.last)
+
+puts "Enabling C&U roles"
+
+server_config = MiqServer.my_server.get_config()
+roles = server_config.config[:server][:role].split(',')
+roles += ['ems_metrics_collector','ems_metrics_coordinator','ems_metrics_processor']
+roles = roles.sort.join(',')
+MiqServer.my_server.set_config(:server=>{:role => roles})
+MiqServer.my_server.get_config().save
```

Comment:
Its not obvious to me my_servet.get_config().save should work.
I think other places that change config do my_server.save (or maybe .save!)
Search set_config...
[I'm on mobile]

Reply:
get_config returns `VMDB::Config`, 

pry's show-source shows
```
From: /home/elad/Projects/manageiq/lib/vmdb/config.rb @ line 55:
Owner: VMDB::Config
Visibility: public
Number of lines: 3

def save(resource = MiqServer.my_server)
  resource.set_config(config)
end
```
so I think you're right, this basically does set_config twice...
The difference between `save` and `save!` on `my_server` is not clear to me. `show-source` shows they're the same, but maybe they're dynamic?

Reply:
I've tried now to read app/models/miq_server/role_management.rb, which seems to operate on app/models/assigned_server_role.rb table, vs app/models/miq_server/configuration_management.rb dealing with config, and I'm giving up :-\
I have no idea how these relate :-(
LGTM if it works...

Reply:
It works, but it's a bit magic.

I spent all day on Thursday looking at `role_management.rb` and trying to use it to enable this functionality, and nothing worked.

Using the settings works, and that's what the UI does too (or at least I think it what it does - the controller is very confusing). When you edit the settings a new item is added to the queue telling all servers to do a configuration reload, and I assume that that's enough for the worker to know it's supposed to collect metrics.